design/authenticators/authn_oidc/oidc_configuration_enhancements.md

```diff
@@ -0,0 +1,228 @@
+- The underlying HTTP client does not honor the `HTTPS_PROXY` environment variable
+- Limited ability to debug OIDC-related HTTP errors
+
+## Solution
+It would be more consistent with other authenticator configs if authn-oidc were to support a `ca-cert` variable 
+in the authenticator policy. This value would be used to inform the HTTP client which CA cert(s) to use to verify 
+the connection with an OIDC provider and/or any proxies that sit in the middle. An authenticator policy featuring 
+this variable may look like:
+```
+	
+- !policy
+  id: conjur/authn-oidc/<service-id>
+  body:
+  - !webservice
+  - !variable token-ttl
+
+  - !variable provider-uri
+  - !variable client-id
+  - !variable client-secret
+
+  # CA Cert [Optional] - Use this to define a custom CA cert or cert chain to verify the connection with the OIDC provider
+  - !variable ca-cert
+
+  # Token TTL [Optional] - Use this to override default
+  - !variable token-ttl
+
+  # URI of Conjur instance
+  - !variable redirect-uri
+
+  # Defines the JWT claim to use as the Conjur identifier
+  - !variable claim-mapping
+
+  # Name [Optional] - Defines Name to display in Conjur-UI
+  - !variable name
+
+  # Provider-Scope [Optional] - Defines claim scope ie: openid email profile
+  - !variable provider-scope
+
+  # Status Webservice for authenticator
+  - !webservice
+    id: status
+    annotations:
+      description: Status service to check that the authenticator is configured correctly
+ 
+  - !group
+    id: operators
+    annotations:
+      description: Group of users who can check the status of the authenticator
+ 
+  - !permit
+     role: !group operators
+     privilege: [ read ]
+     resource: !webservice status
+
+  - !group
+    id: users
+    annotations:
+      description: Group of users who can authenticate using the authn-oidc/<service-id> authenticator
+
+  - !permit
+    role: !group users
+    privilege: [ read, authenticate ]
+    resource: !webservice
+
+#---
+# Need to grant user members to role
+- !grant
+  members:
+  - !user testuser1@mycompany.com
+  role: !group conjur/authn-oidc/<service id>/users
+```
+
+
+## Implementation
+An existing workaround for the CA cert issue mentioned previously involves a user manually updating the 
+OpenSSL truststore in the container to include any custom CA certs. We can leverage this idea in Conjur 
+to do the same thing on the fly. A simple wrapper method which creates a temporary symlink in the OpenSSL 
+truststore to a tempfile containing the certificate content should be sufficient. Cleanup involves 
+removing the symlink, and ensuring that the tempfile has been cleaned up after code execution. Since we 
+are (temporarily) adding the cert to the existing truststore, it should not interfere with concurrent 
+HTTP connections which may rely on the default OpenSSL certs still being available.
+
+An example of what this wrapper method may look like:
```

Comment:
How do we make sure that this wrapper method only injects the certificates we intend and isn't available for use by an attacker?

Reply:
It will only going to inject the cert(s) provided via the authenticator policy, so unless an attacker has the ability to change variable values in the authenticator policy I would think it's a non-issue.

I can't reply to your other comments for some reason but I was curious about what a malicious cert would look like and if that is something we can realistically detect and handle.

Also just pushed an updated doc with a few security considerations I could think of.